torch/testing/_internal/optests/aot_autograd.py

```diff
@@ -54,10 +66,25 @@ def func_no_tensors(args):
 
     compiled_f = compiled_function(
         func_no_tensors, nop, nop, dynamic=dynamic, partition_fn=min_cut_rematerialization_partition)
+
+    any_tensor_requires_grad = pytree.tree_any_only(torch.Tensor, lambda x: x.requires_grad, args)
+    out = wrapper_set_seed(func_no_tensors, args)
+    any_output_requires_grad = pytree.tree_any_only(torch.Tensor, lambda x: x.requires_grad, out)
+    if not any_tensor_requires_grad or not any_output_requires_grad:
```

Comment:
Maybe we should have users tell us explicitly that they are trying to test inference only, so that errors won't be hidden if they forget to do so

Reply:
Sure, I can pass in a `supports_autograd={True, False, "auto"}` argument to make it explicit. For the crossref testing we are doing in D47965247 we want something more automatic to avoid needing to specify what the behavior should be.

Reply:
Flag added